finder-app/writer.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+# validate runtime arguments
+if [ $# -lt 2 ]
+then
+        echo "ERROR: Invalid Number of Arguments."
+        echo "Total number of arguments should be 2."
+        echo "The order of the arguments should be:"
+        echo "\t1)Full path to a file."
+        echo "\t2)Text string to be written to a file."
+        exit 1
+else
+        writefile=$1
+        writestr=$2
+	# extract file directory and file name
+	filedir=`dirname $writefile`
+	filename=`basename $writefile`
+
+        if [ ! -d "$filedir" ]
+        then
+		mkdir -p $filedir
+		if [ $? -eq 1 ]; then
+			echo "ERROR:Unable to create directory."
+			exit 1
+		fi
+	fi
+
+	cd $filedir
```

Comment:
@chchbuff you could skip the `cd` step here and just operate on `$writefile`

Reply:
Thanks for feedback. I thought of doing same later but forgot to modify it. Will update in next assignment.